tests/Module/CheckoutTest.php

```diff
@@ -0,0 +1,578 @@
+						[
+							25061 => 'CZ Zásilkovna do auta',
+						],
+					'isCarDeliveryEnabled'               => true,
+					'isAgeVerificationRequiredByProduct' => false,
+					'cartWeight'                         => 1.0,
+				],
+			'carrier not supporting age verification must be omitted' =>
+				[
+					'expectedRateCount'                  => 0,
+					'carriers'                           =>
+						[
+							DummyFactory::createCarDeliveryCarrier(),
+						],
+					'carriersOptions'                    =>
+						[
+							'packetery_carrier_25061' =>
+								[
+									'id'                  => '25061',
+									'name'                => 'CZ Zásilkovna do auta',
+									'active'              => true,
+									'free_shipping_limit' => null,
+									'weight_limits'       =>
+										[
+											[
+												'weight' => 20.0,
+												'price'  => 234.34,
+											],
+										],
+								],
+						],
+					'productDisallowedRateIds'           => [],
+					'productCategoryDisallowedRateIds'   => [],
+					'allowedCarrierNames'                =>
+						[
+							25061 => 'CZ Zásilkovna do auta',
+						],
+					'isCarDeliveryEnabled'               => true,
+					'isAgeVerificationRequiredByProduct' => true,
+					'cartWeight'                         => 1.0,
+				],
+			'allowed carrier names argument must support null'        =>
+				[
+					'expectedRateCount'                  => 1,
+					'carriers'                           =>
+						[
+							DummyFactory::createCarDeliveryCarrier(),
+						],
+					'carriersOptions'                    =>
+						[
+							'packetery_carrier_25061' =>
+								[
+									'id'                  => '25061',
+									'name'                => 'CZ Zásilkovna do auta',
+									'active'              => true,
+									'free_shipping_limit' => null,
+									'weight_limits'       =>
+										[
+											[
+												'weight' => 20.0,
+												'price'  => 234.34,
+											],
+										],
+								],
+						],
+					'productDisallowedRateIds'           => [],
+					'productCategoryDisallowedRateIds'   => [],
+					'allowedCarrierNames'                => null,
+					'isCarDeliveryEnabled'               => true,
+					'isAgeVerificationRequiredByProduct' => false,
+					'cartWeight'                         => 1.0,
+				],
+		];
+	}
+
+	/**
+	 * @dataProvider rateCreationDataProvider
+	 * @throws \PHPUnit\Framework\MockObject\Exception
+	 */
+	public function testGetShippingRates(
+		int $expectedRateCount,
+		array $carriers,
+		array $carriersOptions,
+		array $productDisallowedRateIds,
+		array $productCategoryDisallowedRateIds,
+		?array $allowedCarrierNames,
+		bool $isCarDeliveryEnabled,
+		bool $isAgeVerificationRequiredByProduct,
+		float $cartWeight
+	): void {
+		$wcProduct = $this->createMock( \WC_Product::class );
+		$wcProduct
+			->method( 'get_price' )
+			->willReturn( '100.00' );
+		$wcProduct
+			->method( 'get_category_ids' )
+			->willReturn( [ 1 ] );
+
+		$cartItem = [
```

Comment:
Please keep "Create Variable" on line 426 and "Use the variable" on line 476 as soon as possible.
<img width="661" alt="Screenshot 2024-08-21 at 11 11 12" src="https://github.com/user-attachments/assets/aa46da07-536f-4908-97d2-3c24fcedc908">